### PR TITLE
Hmeas error

### PR DESCRIPTION
**Pull request by nikivazou**
fix for #631

### CODE REVIEW

File: src/Language/Haskell/Liquid/Constraint/Generate.hs [hunk 1]
```diff
@@ -167,16 +168,17 @@ initEnv info
        f2       <- refreshArgs' $ assm info                  -- assumed refinements      (for imported vars)
        f3       <- refreshArgs' $ vals asmSigs sp            -- assumed refinedments     (with `assume`)
        f40      <- refreshArgs' $ vals ctors sp              -- constructor refinements  (for measures)
+       f5       <- refreshArgs' $ vals inSigs sp            -- assumed refinedments     (with `assume`)
```
Comment (gridaphobe) [inline src/Language/Haskell/Liquid/Constraint/Generate.hs]: the comment here seems to be wrong

**Comment by gridaphobe** [comment]
I'm a bit skeptical of this patch, it looks to me like the following

```haskell
{-@ measure len @-}
{-@ len :: [a] -> {v:Int | v > 0} @-}
len :: [a] -> Int
len [] = 0
len (x:xs) = 1 + len xs
```

would produce a "cannot promote `len`" error, when the error is *not* that we cannot promote `len` but rather that the `Pos` refinement is bogus.

**Comment by nikivazou** [comment]
@gridaphobe the above is just producing a type error

**Comment by nikivazou** [comment]
So, I will fix the comment and merge?

**Comment by gridaphobe** [comment]
My example is effectively giving two types to `len`

```
len :: xs:[a] -> {v:Int | v = len xs}
len :: xs:[a] -> {v:Int | v > 0}
```

which means we should be checking that `len` has the combined type

```
len :: xs:[a] -> {v:Int | v = len xs && v > 0}
```

Can you point out where we're `join`ing the two types? Or how we would end up checking both specifications? 

This is a subtlety that I think should be pointed out in comments because we're either keeping **two** types for `len` around (when I would expect we have an invariant that we only have **one** type for `len` in the environment), or we're joining the two types (in which case we must need to track the types of obligations each conjunct represents).

**Pull request merged**